emulator.c

```diff
@@ -358,6 +396,14 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0xe6: // NOLINT
       printf("ANI ");
+      immediate = cpu_read_mem(cpu, cpu->pc + 1);
+      cpu->a &= immediate;
+      update_zero_flag(cpu, cpu->a);
+      update_sign_flag(cpu, cpu->a);
+      update_parity_flag(cpu, cpu->a);
+      cpu->flags |= FLAG_CY;
```

Comment:
Isn't this the opposite of what you want? The 8080 manual says this instruction clears the carry flag. I believe |= FLAG_CY would always set the carry flag to 1. Why not just update_carry_flag(cpu, false)?